extension/storage/filestorage/client_test.go

```diff
@@ -370,6 +371,88 @@ func BenchmarkClientInitLargeDB(b *testing.B) {
 	}
 }
 
```

Comment:
We should add a test here attempting to do some naive race condition detection. Start some goroutines doing write and read transactions on the db, and then run compaction on it.